From akpm@linux-foundation.org  Thu Aug 19 10:34:11 2010
Message-Id: <201008122048.o7CKmxm5011932@imap1.linux-foundation.org>
Subject: ip2: return -EFAULT on copy_to_user errors
To: mm-commits@vger.kernel.org
Cc: error27@gmail.com, greg@kroah.com, mhw@wittsend.com
From: Dan Carpenter <error27@gmail.com>
Date: Thu, 12 Aug 2010 13:48:59 -0700

From: Dan Carpenter <error27@gmail.com>

copy_to_user() returns the number of bytes remaining but we want to return
a negative error code on errors.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Cc: "Michael H. Warfield" <mhw@wittsend.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/ip2/ip2main.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/char/ip2/ip2main.c
+++ b/drivers/char/ip2/ip2main.c
@@ -2930,6 +2930,8 @@ ip2_ipl_ioctl (struct file *pFile, UINT
 				if ( pCh )
 				{
 					rc = copy_to_user(argp, pCh, sizeof(i2ChanStr));
+					if (rc)
+						rc = -EFAULT;
 				} else {
 					rc = -ENODEV;
 				}
